### ISSUE TITLE
Implement "schema materializer" as soon as it becomes more or less stable

### ISSUE THREAD

**Issue by OlegIlyenko** [sangria-graphql/sangria#21]
https://github.com/graphql/graphql-js/pull/77
https://github.com/graphql/graphql-js/pull/79
https://github.com/graphql/graphql-js/pull/163
https://github.com/graphql/graphql-js/pull/162

`extendSchema` utility: https://github.com/graphql/graphql-js/pull/227

**Comment by OlegIlyenko** [comment]
not at the moment, but maybe later

**Comment by damirv** [comment]
Any updates on this? This would be very convenient! 

By the way: thanks for making this awesome project publicly available!

**Comment by OlegIlyenko** [comment]
Indeed, I also excited about this feature. Ability to reconstruct schema from introspection or IDL can be very useful for different clients and tools. 

I looked at is several times already, but there was still a lot discussion going on. In particular it's not clear yet how to model descriptions in IDL. There are some proposals to introduce annotations or encode it in comments (javadoc style).

But anyways, thanks for reminding me. I think I will review it once again. Maybe it would make sense to implement it now despite the fact that it's bit unstable at the moment, so that people can profit from it.
